CONTRIBUTING.md

```diff
@@ -0,0 +1,201 @@
+
+### Prerequisites
+
+To build TiDB Dashboard you'll need to at least have the following installed:
+
+- `git` - Version control
+- `make` - Build tool (run common workflows)
+- [`golang`](https://golang.org/) - Golang (require 1.13+)
+- [`node.js`](https://nodejs.org/) - Javascript runtime (require 12+)
+
+### Getting the repository
+
+```bash
+git clone https://github.com/pingcap-incubator/tidb-dashboard.git
+cd tidb-dashboard
+# Future instructions assume you are in this directory
+```
+
+### Preparing environment
+
+Optional: set Taobao's npm mirror registry to speed up package downloading if needed
+
+```bash
+npm config set registry https://registry.npm.taobao.org
+```
+
+Install yarn
+
+```bash
+npm install yarn -g
+```
+
+Install Openapi-generator
+
+```bash
+npm install @openapitools/openapi-generator-cli -g
+```
+
+**IMPORTANT**: TiDB Dashboard uses packages from GitHub Packages. you need to configure the ~/.npmrc in order to install these packages.
+
+1. Create a personal access token
+
+    Follow personal ["Settings" -> "Developer settings" -> "Personal access tokens" -> "Generate new token"](https://github.com/settings/tokens/new) path, the "Select scopes" must select `repo` and `read:packages` at least.
+
+2. Edit `~/.npmrc`, add a new line by the following content
+
+```
+//npm.pkg.github.com/:_authToken=[YOUR_PERSONAL_ACCESS_TOKEN]
+```
+
+### Building and running
+
+At this point, you can build and run TiDB Dashboard. 
+
+> Note: TiDB Dashboard need a running TiDB cluster as target, before continue, see [how to start a local TiDB cluster](#starting-a-tidb-cluster).
+
+Build and run back-end server:
+
+```bash
+make
+make run
+# Now tidb-dashboard server is listen on 127.0.0.1:12333
+```
+
+For front-end, you should build API client and start a React develpment server:
+
+```bash
+make swagger_client
+cd ui
+npm run start
+# Now tidb-dashboard UI is available on 127.0.0.1:3000
+```
+
+When you're ready to test out your changes, use the `dev` task. It will lint your code and verify the UI building.
+
+```bash
+make dev
+```
+
+See the [style doc](https://github.com/golang/go/wiki/CodeReviewComments) for details on the conventions.
+
+Please follow this style to make TiDB Dashboard easy to review, maintain, and develop.
+
+### Starting a TiDB cluster
+
+To run TiDB Dashboard, you also need run a local TiDB cluster (at least 1 TiDB server), here we introduce how to start a local TiDB cluster by binary deployment. Also, you can use other ways like [docker-compose](https://github.com/pingcap/tidb-docker-compose#quick-start) or [TiUP](https://tiup.io).
```

Comment:
I think we can put more texts for TiUP. It is much more easy to use TiUP to bootstrap a cluster locally.

Reply:
TiUP is not ready now, We use it as default way in the future.

Reply:
Its playground feature is enough for our development